scripts/repo.js

```diff
@@ -0,0 +1,193 @@
+      ['The UiAutomator2 Driver', 'uiautomator2.md'],
+      ['The Espresso Driver', 'espresso.md'],
+      ['Windows App Testing', 'windows-app-testing.md']]],
+    ['Advanced', ['advanced-concepts',
+      ['Migrating to Appium 1.0', 'migrating-to-1-0.md'],
+      ['Migrating to XCUITest', 'migrating-to-xcuitest.md'],
+      ['Automating Hybrid Apps', 'hybrid.md'],
+      ['Using Selenium Grid with Appium', 'grid.md'],
+      ['Chromedriver Troubleshooting', 'chromedriver.md'],
+      ['Cross-domain iframes', 'cross-domain-iframes.md'],
+      ['Using ios-webkit-debug-proxy', 'ios-webkit-debug-proxy.md'],
+      ['Using instruments-without-delay on Xcode 7', 'iwd_xcode7.md'],
+      ['Using a custom WDA server', 'wda-custom-server.md'],
+      ['The Event Timings API', 'event-timings.md'],
+      ['The Settings API', 'settings.md']]],
+    ['Contributing', ['contributing-to-appium',
+      ['Running Appium from Source', 'appium-from-source.md'],
+      ['Developer Overview', 'developers-overview.md'],
+      ['Standard Gulp Commands', 'gulp.md'],
+      ['Appium Style Guide', 'style-guide-2.0.md'],
+      ['How to Write Docs', 'how-to-write-docs.md'],
+      ['Appium Package Structure', 'appium-packages.md'],
+      ['Credits', 'credits.md']]]
+  ], cn: [
+  ]
+};
+
+async function unzipStream (readstream, pathToUnzipped) {
+  return new B((resolve, reject) => {
+    readstream.pipe(unzip.Extract({
+      path: pathToUnzipped,
+    }));
+    readstream.on('end', resolve);
+    readstream.on('close', reject);
+  });
+}
+
+async function getRepoDocs (owner, repo, branch='master') {
+  // Pull down a zipball of Appium repository
+  const github = new Github({
+    Promise: B,
+  });
+  const githubBranch = await github.repos.getBranch({owner, repo, branch});
+  const ref = githubBranch.data.commit.sha;
+  const archive = await request(`https://api.github.com/repos/${owner}/${repo}/zipball/${ref}`, {
+    headers: {
+      'User-Agent': 'appium',
+    },
+    encoding: null,
+  });
+
+  // Unzip it to a temporary directory
+  const tempdir = await tempDir.openDir();
+  const pathToZip = path.resolve(tempdir, 'appium.zip');
+  await fs.writeFile(pathToZip, archive);
+  const readStream = nodeFS.createReadStream(pathToZip);
+
+  const pathToUnzipped = path.resolve(tempdir, 'appium', branch);
+  await mkdirp(pathToUnzipped);
+
+  await unzipStream(readStream, pathToUnzipped);
+
+  // Get a reference to the path to the docs
+  const pathToAppiumFiles = await fs.readdir(pathToUnzipped);
+  const pathToDocs = path.resolve(pathToUnzipped, pathToAppiumFiles[0], 'docs');
+
+  // Do some treating on the docs to get it working correctly
+  await alterDocs(pathToDocs);
+
+  return pathToDocs;
+}
+
+/**
+ * Adjust the contents of the docs to fit proper MkDocs format
+ *  - rename README.md to index.md
+ * @param {*} pathToDocs
+ */
+async function alterDocs (pathToDocs) {
+  for (let file of await fs.readdir(pathToDocs)) {
```

Comment:
I still think this manual stuff (here and in `applyTabs`) ought to be replaced with `glob`.